src/Lock.php

```diff
@@ -85,34 +85,107 @@ public static function showForItem(CommonDBTM $item)
+                $lockable_object = new $lockable_itemtype();
+                $query  = [
+                    'SELECT' => $lockedfield->getTable() . ".*",
+                    'FROM'   => $lockedfield->getTable(),
+                    'LEFT JOIN' => [
+                        getTableForItemType($lockable_itemtype)   => [
+                            'FKEY'   => [
+                                $lockedfield->getTable()  => 'items_id',
+                                getTableForItemType($lockable_itemtype)   => 'id'
+                            ]
+                        ]
+                    ],
+                    'WHERE'  => [
+                        'OR' => [
+                            [
+                                $lockedfield->getTable() . '.itemtype'  => $lockable_itemtype,
+                                $lockedfield->getTable() . '.items_id'  => new \QueryExpression(getTableForItemType($lockable_itemtype) . '.id')
+                            ], [
+                                $lockedfield->getTable() . '.itemtype'  => $lockable_itemtype,
+                                $lockedfield->getTable() . '.is_global' => 1
+                            ]
+                        ]
+                    ]
+                ];
+
+                if (
+                    is_a($lockable_itemtype, CommonDBConnexity::class, true)
+                    && (
+                        (property_exists($lockable_object, 'items_id') && $lockable_object::$items_id == "items_id")
+                        || (property_exists($lockable_object, 'items_id_1') && $lockable_object::$items_id_1 == "items_id")
+                        || (property_exists($lockable_object, 'items_id_2') && $lockable_object::$items_id_2 == "items_id")
+                    )
+                ) {
+                    $query['WHERE'][] = [
+                        getTableForItemType($lockable_itemtype) . '.itemtype' => $itemtype,
+                        getTableForItemType($lockable_itemtype) . '.items_id' => $ID
+                    ];
+                } else {
+                    if (
+                        property_exists($lockable_object, 'items_id')
+                        && getForeignKeyFieldForItemType($itemtype) == $lockable_object::$items_id
+                    ) {
+                        $query['WHERE'][] = [getTableForItemType($lockable_itemtype) . '.' . $lockable_object::$items_id => $ID];
+                    } elseif (property_exists($lockable_object, 'items_id_1') && getForeignKeyFieldForItemType($itemtype) == $lockable_object::$items_id_1) {
+                        $query['WHERE'][] = [getTableForItemType($lockable_itemtype) . '.' . $lockable_object::$items_id_1 => $ID];
+                    } else {
+                        // ::$items_id != getForeignKeyFieldForItemType($itemtype)
+                        // ex: get ComputerVirtualMachine from NetworkEquipement
+                        continue;
+                    }
+                }
```

Comment:
You could probably use result of `CommonDBConnexity::getSQLCriteriaToSearchForItem()` here.
```suggestion
                if (is_a($lockable_itemtype, CommonDBConnexity::class, true)) {
                    $connexity_criteria = $lockable_itemtype::getSQLCriteriaToSearchForItem($itemtype, $ID);
                    if ($connexity_criteria === null) {
                        continue;
                    }
                    $query['WHERE'][] = $connexity_criteria['WHERE'];
                }
```

Reply:
does not work with this method, the generated ```where``` clause contains ambiguous column

```sql
  Error: Column 'items_id' in where clause is ambiguous
  Backtrace :
  src/DBmysqlIterator.php:110                        
  src/DBmysql.php:1048                               DBmysqlIterator->execute()
  src/Lock.php:158                                   DBmysql->request()
  src/Lock.php:660                                   Lock::showForItem()
  src/CommonGLPI.php:689                             Lock::displayTabContentForItem()
  ajax/common.tabs.php:113                           CommonGLPI::displayStandardTab()
```

Reply:
Maybe the method could be updated to user `table.field` instead of only `field` on its `WHERE` conditions.

Reply:
I think it's going to be quite difficult

```sql
  Error: Unknown column 'glpi_computers.computers_id' in 'where clause'
  Backtrace :
  src/DBmysqlIterator.php:110                        
  src/DBmysql.php:1048                               DBmysqlIterator->execute()
  src/Lock.php:183                                   DBmysql->request()
  src/Lock.php:684                                   Lock::showForItem()
  src/CommonGLPI.php:689                             Lock::displayTabContentForItem()
  ajax/common.tabs.php:113                           CommonGLPI::displayStandardTab()
  {"user":"2@stanislas-asus-desktop"} 
```